### ISSUE TITLE
Support for wxWidgets X11 target with NX11?

### ISSUE THREAD

**Issue by ghost** [ghaerr/microwindows#23]
Is there any chance more functionality could be added so wxWidgets X11 target could work via NX11? It'd be a nice alternative to FLTK. The microwindows code in wxWidgets doesn't really support anything, so targeting the X11 target would be better.

**Comment by georgp24** [comment]
The Microwindows support of wxWidgets was implemented in a subset of wxWidgets. Not sure if that ever worked properly. 
To answer your question one would have to try to compile wxWidgets with NX11 and see how far you get.

**Comment by ghaerr** [comment]
Yes, we'd be happy to help.  Are you associated with wxWidgets or just thinking about moving your own project moving forward with wxWidgets?

I was looking at wxWidgets a couple of weeks ago and though this might be interesting. Can you describe the microwindows code already in wxWdigets? I suppose its from quite some time ago.

The latest NX11 compatibility library is pretty good, it implements every X11 call from FLTK and runs with no changes in the FLTK v1.3.4-2 source (other than a required commenting out of dynamically loading XRandR which isn't needed or supported).

The first step would be to configure wxWidgets on an X11 target, but then linking either dynamically or statically with -lNX11 and -lnano-X. I recommend static initially, then the linker will report unimplemented X11 routines. Microwindows should be configured using say, src/Configs/config.linux-X11 as the src/config file, and the libraries will be built in src/lib. There may be some unimplemented stubs used in src/nx11/stubs.c, but these will printf a debug message when called if the config file option DEBUG=Y is set.

I'm happy to help implement the required additional NX11 routines to get wxWidgets running.

**Comment by exdeus7** [comment]
Thanks for the reply. I'm not associated with WxWidgets. I work in the embedded linux realm and have recently evaluated quite a few GUI frameworks for embedded use. I think WxWidgets could work really well there.

I'll follow the steps you listed and let you know how it goes.

Thanks!

**Comment by nwrkbiz** [comment]
> Thanks for the reply. I'm not associated with WxWidgets. I work in the embedded linux realm and have recently evaluated quite a few GUI frameworks for embedded use. I think WxWidgets could work really well there.
> 
> I'll follow the steps you listed and let you know how it goes.
> 
> Thanks!

Did you check out FLTK aswell? In my opinion FLTK would be a better choice for embedded systems. (depending on your use case)

**Comment by progfrosty** [comment]
Would love to see wxWidgets working properly with NX11. What is the status, I notice a few commits last year concerning wxWidgets.

**Comment by ghaerr** [comment]
A long time ago, someone added direct Nano-X support into wxWidgets, but apparently nothing has happened since.

I took a quick look at wxWidgets when this issue was opened, but don't have the time to get into it personally at this time. I'm not sure whether it would be better to try linking to it with NX11, or go with direct support. I suppose it would be a lot easier to start with NX11.

**Comment by georgp24** [comment]
I got a very old version of wxWidgets to work with NX11. However, in any case this uses the wxWidgets for X11 version and this branch of wxWidgets has not been maintained for a long time. It also does not work equally well as the other platform branches wxWidgets supports.
